Use sane defaults for zwave

especially-relative opened the pull request:
**Description:**

Use sane default if libopenzwave is installed. In most cases this will mean that the zwave config path will not need to e manually specified.

Verify write access to the usb device before opening. If write access is missing, print an error indicating the current user, and path of usb device.

**Related issue (if applicable):** #1884, verifies https://github.com/home-assistant/home-assistant.io/pull/414 with readable error message.

**Example entry for `configuration.yaml` (if applicable):**

``` yaml

```

**Checklist:**

If code communicates with devices:
- [X] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
- [X] New dependencies have been added to the `REQUIREMENTS` variable ([example](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L16)).
- [X] New dependencies are only imported inside functions that use them ([example](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L51)).
- [n/a] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
- [n/a] New files were added to `.coveragerc`.

If the code does not interact with devices:
- [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
- [ ] Tests have been added to verify that the new code works.

Use sane default if libopenzwave is installed.

Verify write access to the usb device before opening.

balloob commented on homeassistant/components/zwave.py:
```diff
@@ -17,6 +20,14 @@
+    DEFAULT_ZWAVE_CONFIG_PATH = os.path.join(os.path.dirname(
+        libopenzwave.__file__), 'config')
+except NameError:
+    DEFAULT_ZWAVE_CONFIG_PATH = os.path.join(sys.prefix, 'share',
```
No need to default anything as without openzwave it's not going to work. 

Should we detect if openzwave doesn't exist in the setup and link to instructions?

danieljkemp replied:
The idea is to default to `libopenzwave.__file__` as the config location if one is not specified. This will be the correct location except in edge cases where the user has modified/updated the libopenzwave config manually. The goal is for elimiating `config_path: ` as a required yaml entry except for those rare edge cases.

You're right about the failure mode however, if libopenzwave is not found, we should exit the component then and there.

balloob commented on homeassistant/components/zwave.py:
```diff
@@ -17,6 +20,14 @@
+                                             'python-openzwave', 'config')
+
 DOMAIN = "zwave"
 REQUIREMENTS = ['pydispatcher==2.0.5']
```
We might drop this as its installed as part of openzwave?

balloob commented on homeassistant/components/zwave.py:
```diff
@@ -53,7 +63,7 @@
 COMMAND_CLASS_BATTERY = 128
 COMMAND_CLASS_ALARM = 113  # 0x71
-COMMAND_CLASS_THERMOSTAT_MODE = 64  # 0x40
+COMMAND_CLASS_THERMOSTAT_SETPOINT = 67  # 0x43
```
Seems unrelated?

danieljkemp replied:
Not sure what happened there with the command classes, I didn't actually touch those myself. Git:1, Me: 0.

danieljkemp commented on homeassistant/components/zwave.py:
```diff
@@ -183,9 +193,18 @@ def setup(hass, config):
+        _LOGGER.error("Current user '%s' does not have write permissions to"
+                      "usb device '%s'", getpass.getuser(), usb_stick_path)
+        return
+
```
Turns out libopenzwave already does this. I'll leave it to them, unless we want to try for a more verbose error than what they return.

danieljkemp commented:
Updated PR, fixed the clobbering of thermostat command classes, left write access checking to libopenzwave, and changed where imports are being done.

JshWright commented:
Looks good to me

SEJeff commented:
:+1:

balloob commented on homeassistant/components/zwave.py:
```diff
@@ -175,10 +172,14 @@ def setup(hass, config):
     # pylint: disable=global-statement, import-error
     global NETWORK
 
+    import libopenzwave
```
Since we can't install this using `REQUIREMENTS`, I would suggest we wrap this and print a proper warning to the user:
```python
try:
    import libopenzwave
except ImportError:
    _LOGGER.error("You are missing required dependency Python Open Z-Wave. Please follow instructions at X")
    return False
```

balloob commented:
Looks good 🐬 Added 1 comment that would improve the experience a bit more.

Commit:
Make zwave component user sane defaults.

Change the zwave config default path to be valid in all but rare
edge cases. This will let that config value be truely optional.

Also check that libopenzwave is installed and print a warrning if
it cannot be found pointing at the site.

balloob commented:
💃 Looking good 🐬
